gst-play/gst-play.c

```diff
@@ -110,6 +110,222 @@ buffering_cb (GstPlayer * player, gint percent, GstPlay * play)
+    else
+      print_subtitle_info ((GstPlayerSubtitleInfo*)stream);
+  }
+}
+
+static void
+print_all_video_stream (GstPlayerMediaInfo *media_info)
+{
+  GList *list = NULL, *l;
+
+  list = gst_player_get_video_streams (media_info);
+  if (!list)
+    return;
+
+  g_print ("All video streams\n");
+  for (l = list; l != NULL; l = l->next) {
+    GstPlayerVideoInfo  *info = (GstPlayerVideoInfo*) l->data;
+    GstPlayerStreamInfo *sinfo = (GstPlayerStreamInfo*) info;
+    g_print (" %s_%d #\n", gst_player_stream_info_get_stream_type_nick (sinfo),
+            gst_player_stream_info_get_stream_index (sinfo));
+    print_video_info (info);
+  }
+}
+
+static void
+print_all_subtitle_stream (GstPlayerMediaInfo *media_info)
+{
+  GList *list = NULL, *l;
+
+  list = gst_player_get_subtitle_streams (media_info);
+  if (!list)
+    return;
+
+  g_print ("All subtitle streams:\n");
+  for (l = list; l != NULL; l = l->next) {
+    GstPlayerSubtitleInfo  *info = (GstPlayerSubtitleInfo*) l->data;
+    GstPlayerStreamInfo *sinfo = (GstPlayerStreamInfo*) info;
+    g_print (" %s_%d #\n", gst_player_stream_info_get_stream_type_nick (sinfo),
+    gst_player_stream_info_get_stream_index (sinfo));
+    print_subtitle_info (info);
+  }
+}
+
+static void
+print_all_audio_stream (GstPlayerMediaInfo *media_info)
+{
+  GList *list = NULL, *l;
+
+  list = gst_player_get_audio_streams (media_info);
+  if (!list)
+    return;
+
+  g_print ("All audio streams: \n");
+  for (l = list; l != NULL; l = l->next) {
+    GstPlayerAudioInfo  *info = (GstPlayerAudioInfo*) l->data;
+    GstPlayerStreamInfo *sinfo = (GstPlayerStreamInfo*) info;
+    g_print (" %s_%d #\n", gst_player_stream_info_get_stream_type_nick (sinfo),
+              gst_player_stream_info_get_stream_index (sinfo));
+    print_audio_info (info);
+  }
+}
+
+static void
+print_current_tracks (GstPlay *play)
+{
+  GstPlayerAudioInfo *audio = NULL;
+  GstPlayerVideoInfo *video = NULL;
+  GstPlayerSubtitleInfo *subtitle = NULL;
+
+  g_print ("Current video track: \n");
+  video = gst_player_get_current_video_track (play->player);
+  print_video_info (video);
+
+  g_print ("Current audio track: \n");
+  audio = gst_player_get_current_audio_track (play->player);
+  print_audio_info (audio);
+
+  g_print ("Current subtitle track: \n");
+  subtitle = gst_player_get_current_subtitle_track (play->player);
+  print_subtitle_info (subtitle);
+
+  if (audio)
+    g_object_unref (audio);
+
+  if (video)
+    g_object_unref (video);
+
+  if (subtitle)
+    g_object_unref (subtitle);
+}
+
+static void
+print_media_info (GstPlayerMediaInfo *media_info)
+{
+  if (!media_info)
+    return;
+
+  print_all_stream_info (media_info);
+  g_print ("\n");
+  print_all_video_stream (media_info);
+  g_print ("\n");
+  print_all_audio_stream (media_info);
+  g_print ("\n");
+  print_all_subtitle_stream (media_info);
+}
+
+static void
+media_info_cb (GstPlayer *player, GstPlayerMediaInfo *info, GstPlay *play)
+{
+}
```

Comment:
How about if I print media information once ?

e.g something like
static void media_info_cb (GstPlayer *player, GstPlayerMediaInfo *inf, GstPlay *play)
{
  static int once = 0;
  if (!once) {
   call function to print the media information
   once=1;
  }
}